pkg/transport/managed_transport.go

```diff
@@ -75,22 +75,27 @@ type ManagedTransport struct {
 	once sync.Once
 	wg   sync.WaitGroup
 
-	remoteAddrs []string
+	remoteAddr string
+
+	afterClosedMu sync.RWMutex
+	afterClosed   TPCloseCallback
 }
 
 // NewManagedTransport creates a new ManagedTransport.
-func NewManagedTransport(n *snet.Network, dc DiscoveryClient, ls LogStore, rPK cipher.PubKey, netName string) *ManagedTransport {
+func NewManagedTransport(n *snet.Network, dc DiscoveryClient, ls LogStore, rPK cipher.PubKey,
```

Comment:
What do you think about passing a structure instead of the argument list here? The list became huge.